File: project3/modules/floor.cpp
#include <iostream>
#include <floor.h>

#define NUMCLASS 6

Floor::Floor(const unsigned numberFloor,const unsigned sizeClass):
    floor(numberFloor){
    std::cout<<"A new Floor with number " <<floor<<" is being created\n";
    classes=new TheClass*[NUMCLASS];
    for (unsigned i=0;i<NUMCLASS;i++){
        if(i<3)
            classes[i]= new TheClass(floor,i,sizeClass);
        else
            classes[i]= new TheClass(floor,i,sizeClass);
    }
    corridor=new Corridor(numberFloor);
}


Floor::~Floor(){
    for (int i=0;i<NUMCLASS;i++)
        delete classes[i];
    delete[] classes;
    delete corridor;
    std::cout<<"A Floor with number " <<floor<<" to be destroyed\n";
}

TheClass& Floor::getClass(int index) const{
    return *(classes[index]);
}

Corridor& Floor::getCorridor() const{
    return *corridor;
}

void Floor::print()const{
    std::cout<<"Floor number "<<floor<<" contains:\n";
    for(int i=0;i<NUMCLASS;i++)
        classes[i]->print();
    corridor->print();

}

File: project4/src/schoolComplex/Floor.java
package schoolComplex;

import java.util.Vector;

public class Floor {
        
    private int floor;
    private Vector<TheClass> classes;
    private Corridor corridor;
    
    public Floor(int floorNumber,int sizeClass){
        classes = new Vector<TheClass>(6);
        for (int i=0;i<6;i++)
            classes.add(new TheClass(floorNumber,i,sizeClass));
        corridor= new Corridor(floorNumber);
        floor=floorNumber;
    }    
    public TheClass getClass(int index){
        return classes.get(index);
    }
    public Corridor getCorridor(){
        return corridor;
    }
    
    public void print(){
        System.out.println("Floor (number:"+Integer.toString(floor)+")");
        for (int i=0;i<6;i++)
            classes.get(i).print();
        corridor.print();
    }
    public void operate(int hours){
        for (int i=0;i<6;i++)
            classes.get(i).operate(hours);
    }
}

